src/zarr/v3/sync.py

```diff
@@ -57,30 +65,31 @@ def sync(coro: Coroutine, loop: Optional[asyncio.AbstractEventLoop] = None):
         # NB: if the loop is not running *yet*, it is OK to submit work
         # and we will wait for it
         loop = _get_loop()
-    if loop is None or loop.is_closed():
+    if not isinstance(loop, asyncio.AbstractEventLoop):
+        raise TypeError(f"loop cannot be of type {type(loop)}")
+    if loop.is_closed():
         raise RuntimeError("Loop is not running")
     try:
         loop0 = asyncio.events.get_running_loop()
         if loop0 is loop:
-            raise NotImplementedError("Calling sync() from within a running loop")
+            raise SyncError("Calling sync() from within a running loop")
```

Comment:
I believe this was a bug before as `NotImplementedError` inherits from `RuntimeError` so this exception never raised.